src/client/common/process/baseDaemon.ts

```diff
@@ -177,7 +178,7 @@ export abstract class BasePythonDaemon {
         return Object.keys(options).every((item) => daemonSupportedSpawnOptions.indexOf(item as any) >= 0);
     }
     protected sendRequestWithoutArgs<R, E, RO>(type: RequestType0<R, E, RO>): Thenable<R> {
-        if (this.proc && typeof this.proc.exitCode !== 'number') {
+        if (this.isAlive && this.proc && typeof this.proc.exitCode !== 'number') {
```

Comment:
While running the tests locally found that there were some other unhandled exceptions.
Basically after the tests we kill the daemon, however some async code is still attempting to write to the daemon.

This ensures we handle that situation & thus prevent logging or unhandled errors (i.e. spurious errors being logged)

Reply:
Nice. Think that I'd seen those errors cluttering up the logs before.